Math/NumberTheory/Moduli/Sqrt.hs

```diff
@@ -33,9 +36,7 @@ import Math.NumberTheory.Primes.Testing.Probabilistic (isPrime)
 import Math.NumberTheory.Utils (shiftToOddCount, splitOff)
 
 -- | Extended flag structure to check if square root is computed in finite field.
-data FieldCharacterictic = Invalid
-                         | Prime
-                         | PrimePower Integer Int
+data FieldCharacterictic = FieldCharacterictic Integer Int
```

Comment:
Could we have `Prime Integer` instead of `Integer` here? To make `FieldCharacterictic` correct by construction.

Reply:
I'll try